assets/app/scripts/controllers/create.js

```diff
@@ -47,33 +119,198 @@ angular.module('openshiftConsole')
+        // looking at status tags.
+        var specTags = {};
+        if (imageStream.spec && imageStream.spec.tags) {
+          angular.forEach(imageStream.spec.tags, function(tag) {
+            if (tag.annotations && tag.annotations.tags) {
+              specTags[tag.name] = tag.annotations.tags.split(/\s*,\s*/);
+            }
+          });
         }
-      };
 
-      angular.forEach(projectTemplates, categorizeTemplates);
-      angular.forEach(openshiftTemplates, categorizeTemplates);
+        // Loop over status tags to categorize the images.
+        angular.forEach(imageStream.status.tags, function(tag) {
+          var imageStreamTag = tag.tag;
+          var category;
+          var categoryTags = specTags[imageStreamTag] || [];
+          var image = {
+            imageStream: imageStream,
+            imageStreamTag: imageStreamTag,
+            name: imageStream.metadata.name + ":" + imageStreamTag,
+            version: imageStreamTagAnnotationFilter(imageStream, 'version', imageStreamTag),
+            categoryTags: categoryTags
+          };
+          if (categoryTags.indexOf("builder") >= 0) {
+            // Add the builder image to its category.
+            category = getCategory(categoryTags);
+            if (!buildersByCategory[category]) {
+              buildersByCategory[category] = [];
+            }
 
-      Logger.info("instantApps", $scope.instantApps);
-      Logger.info("otherTemplates", $scope.otherTemplates);
+            buildersByCategory[category].push(image);
+          } else {
+            // Group non-builder images separately so we can hide them by default.
+            nonBuilderImages.push(image);
+          }
+        });
+      });
     }
 
-    $scope.createFromSource = function() {
-      if($scope.from_source_form.$valid) {
-        var createFromSourceURL = createFromSourceURLFilter($scope.projectName, $scope.from_source_url);
-        $location.url(createFromSourceURL);
+    function categorizeTemplates(templates) {
+      angular.forEach(templates, function(template) {
+        var tags = tagsFilter(template);
+        var category = getCategory(tags);
+        if (!templatesByCategory[category]) {
+          templatesByCategory[category] = [];
+        }
+
+        templatesByCategory[category].push(template);
+      });
+    }
+
+    function getCategory(tags) {
+      var i, j;
+
+      // Find the first matching category tag in tags.
+      for (i = 0; i < $scope.categoryTags.length; i++) {
+        for (j = 0; j < tags.length; j++) {
+          if (tags[j].toLowerCase() === $scope.categoryTags[i]) {
+            return tags[j];
+          }
+        }
       }
-    };
+
+      return "";
+    }
+
+    function updateState() {
+      // Have we finished loading all of the templates and image streams in
+      // both the project and openshift namespaces? If undefined, they're not
+      // loaded.
+      $scope.loaded =
+        projectTemplates &&
```

Comment:
I'm not sure we want to. We'll show any data that's there. If we set things to empty maps, it might result in an incorrect message saying you have no images or templates. You do get an error alert from DataService.